### PR TITLE
Refactor collectors documentation

### PR DESCRIPTION
**Pull request by joelhans**
##### Summary

Fixes: #7878

@ilyam8's work on #7878 inspired me to take on the major project of refactoring the collectors documentation to help users better understand what they actually want to know: **How to collect metrics from X app/service**. Aside from that, I'm piggybacking off of @ilyam8's work to create a single source of truth of what collectors we have and support instead of having it duplicated in many different places.

There's a lot of work to be done still but I wanted some to be able to see what I'm doing.

The main goals:

- [x] Put key information about enabling `collectors/README.md` and `collectors/QUICKSTART.md`
- [x] Put up-to-date list of collectors in `collectors/LIST.md`
- [x] Remove `docs/Add-more-charts-to-netdata.md` and point link from dashboard to new collectors docs
- [x] Help clarify what collectors, plugins, orchestrators, and modules are, and emphasize that most users really don't need to know
- [x] Push complex information about writing collectors/APIs deeper into the docs structure
- [x] Begin work across docs emphasizing the word "collector" over "plugin" or "module" when referencing how users will get metrics from new sources

TODOs: 

- [x] Edit all the new and edited content for grammar and typos
- [x] Write the quickstart guide (`QUICKSTART.md`)
- [x] Add collector troubleshooting details to `REFERENCE.md`
- [x] Sync `README.md` and `docs/what-is-netdata.md`

##### Component Name

docs
collectors/

### CODE REVIEW

**Comment by ilyam8** [comment]
> Organize collectors by their type, not their plugin/orchestrator

I think it should be done later, it will take a lot of time to group them by type (what is type btw? 😄).

Grouping by orchestrator is ok too.

**Comment by cakrit** [comment]
> I think it should be done later, it will take a lot of time to correctly group them by type (what is type btw? 😄).
> 
> Grouping by orchestrator is ok too.

Nope, we need them aligned to https://github.com/netdata/ac-netdata/tree/master/src/data/integrations

File: collectors/REFERENCE.md [hunk 1]
```diff
@@ -40,7 +41,41 @@ There are three types of plugins:
...
+./go.d.plugin -d -m <MODULE_NAME>
+
+# Python orchestrator (python.d.plugin)
+./python.d.plugin <MODULE_NAME> debug trace
+
+# Node orchestrator (node.d.plugin)
+./node.d.plugin debug 1 <MODULE_NAME>
+
+# Bash orchestrator (bash.d.plugin)
+./charts.d.plugin debug 1 <MODULE_NAME>
```
Comment (ilyam8) [inline collectors/REFERENCE.md]: not sure that thing have per module debug mode, same for `node.d.plugin` 😄

Comment (ilyam8) [reply collectors/REFERENCE.md]: it seems bash has no (at least `debug 1 name` doesnt work)

```
st@DEB10-1:/opt/netdata/usr/libexec/netdata/plugins.d$ sudo su
root@DEB10-1:/opt/netdata/usr/libexec/netdata/plugins.d# ./charts.d.plugin debug 1 nut
2020-02-10 09:58:29: charts.d: INFO: main: started from './charts.d.plugin' with options: debug 1 nut
2020-02-10 09:58:29: charts.d: INFO: main: Configuration file '/opt/netdata/usr/lib/netdata/conf.d/charts.d.conf' loaded.
2020-02-10 09:58:29: charts.d: WARNING: main: Configuration file '/opt/netdata/etc/netdata/charts.d.conf' not found.
2020-02-10 09:58:29: charts.d: INFO: apache: is disabled. Add a line with apache=force in '/opt/netdata/etc/netdata/charts.d.conf' to enable it (or remove the line that disables it).
2020-02-10 09:58:29: charts.d: DEBUG: ap: is enabled for auto-detection.
2020-02-10 09:58:29: charts.d: DEBUG: apcupsd: is enabled for auto-detection.
```
___

`node.d.plugin` HAS per module debug mode

Comment (joelhans) [reply collectors/REFERENCE.md]: According to [the docs](https://docs.netdata.cloud/collectors/charts.d.plugin/#debugging-your-collectors), `charts.d.plugin` _should_ be able to use the syntax, but you're right that it doesn't actually work that way. No matter what module you put in it returns output from all modules.

Comment (ilyam8) [reply collectors/REFERENCE.md]: ok, i am wrong, it works, but 

- it support only 1 module in debug mode (`./charts.d.plugin debug 1 X`)
- it **loads all** modules (configs, runs checks for all of them)
- and only after it keeps running the one that is in debug

Comment (ilyam8) [reply collectors/REFERENCE.md]: that is debug code

https://github.com/netdata/netdata/blob/d3fa405141a8d2e6689c2dedb4bedb7cf9120044/collectors/charts.d.plugin/charts.d.plugin.in#L546-L558

load all modules, run check method for all of them code before it

https://github.com/netdata/netdata/blob/d3fa405141a8d2e6689c2dedb4bedb7cf9120044/collectors/charts.d.plugin/charts.d.plugin.in#L489-L534

**Comment by joelhans** [comment]
Another benefit of this PR is that it brings linter errors to a measly 5850!

**Pull request merged**